src/sharing/SharingScreen.js

```diff
@@ -0,0 +1,467 @@
+    const narrow = streamNarrow(stream);
+
+    return (
+      <>
+        <ScrollView style={styles.wrapper} keyboardShouldPersistTaps>
+          <View style={styles.container}>
+            {share.type === 'image' && (
+              <Image
+                source={{ uri: share.sharedImageUri }}
+                width={200}
+                height={200}
+                style={styles.imagePreview}
+              />
+            )}
+            <AnimatedScaleComponent visible={isStreamFocused}>
+              <StreamAutocomplete filter={stream} onAutocomplete={this.handleStreamAutoComplete} />
+            </AnimatedScaleComponent>
+            <Input
+              placeholder="Stream"
+              value={stream}
+              onChangeText={this.handleStreamChange}
+              onFocus={this.focusStream}
+              onChange={this.focusStream}
+              onBlur={this.blurStream}
+              selectTextOnFocus
+            />
+            <View />
+            <AnimatedScaleComponent visible={isTopicFocused}>
+              <TopicAutocomplete
+                text={topic}
+                onAutocomplete={this.handleTopicAutoComplete}
+                isFocused={isTopicFocused}
+                narrow={narrow}
+              />
+            </AnimatedScaleComponent>
+            <Input
+              placeholder="Topic"
+              value={topic}
+              onFocus={this.focusTopic}
+              onBlur={this.blurTopic}
+              onChangeText={this.handleTopicChange}
+              editable={stream !== ''}
+            />
+            <Input
+              placeholder="Message"
+              value={message}
+              onChangeText={this.handleMessageChange}
+              multiline
+            />
+          </View>
+        </ScrollView>
+        <View style={styles.actions}>
+          <ZulipButton onPress={this.finishShare} style={styles.button} secondary text="Cancel" />
+          <ZulipButton
+            style={styles.button}
+            onPress={this.handleSend}
+            text="Send"
+            disabled={!this.isSendButtonEnabled()}
+          />
+        </View>
+      </>
+    );
+  }
+}
+
+type PmScreenProps = $ReadOnly<{|
+  share: SharedData,
+  dispatch: Dispatch,
+  auth: Auth,
+|}>;
+
+type PmScreenState = $ReadOnly<{|
+  users: User[],
+  message: string,
+  choosingRecipients: boolean,
+|}>;
+
+class PmScreen extends React.Component<PmScreenProps, PmScreenState> {
+  constructor(props) {
+    super(props);
+    const { share } = props;
+    this.state = {
+      users: [],
+      message: share.type === 'text' ? share.sharedText : '',
+      choosingRecipients: false,
+    };
+  }
+  toggleChoosingRecipients = () => {
+    this.setState(state => ({ choosingRecipients: !state.choosingRecipients }));
+  };
+  handleChooseRecipients = (users: Array<User>) => {
+    this.setState({ users });
+    this.toggleChoosingRecipients();
+  };
+  finishShare() {
```

Comment:
blank lines between all these definitions, please -- it makes it a lot easier to read :slightly_smiling_face: